lib/wallaroo/core/sink/connector_sink/connector_sink.pony

```diff
@@ -217,25 +247,85 @@ actor ConnectorSink is Sink
     msg_uid: MsgId, frac_ids: FractionalMessageId, i_seq_id: SeqId,
     latest_ts: U64, metrics_id: U16, worker_ingress_ts: U64)
   =>
+    ifdef "trace" then
+      @printf[I32]("Rcvd msg at ConnectorSink\n".cstring())
+    end
+    if _twopc_state is cp.TwoPCFsm2Abort then
+      // TODO I think that this is harmless, but double-check.
+      // We've done prepare_for_rollback before getting here.
+      // The next step is rollback().  Anything that we send to
+      // the sink is going to be undone.  Aside from the wasted
+      // effort, is there any harm done?  Is it worth the effort
+      // of trying to switch to NormalSinkMessageProcessor and
+      // sync _twopc_state=TwoPCFsmStart in only 1 behavior call?
+      // And can we do that without adding more bugs?
+      None
+    elseif not (_twopc_state is cp.TwoPCFsmStart) then
+      @printf[I32]("2PC: ERROR: _twopc_state = %d\n".cstring(), _twopc_state())
+      Fail()
+    end
+
     var receive_ts: U64 = 0
     ifdef "detailed-metrics" then
-      receive_ts = WallClock.nanoseconds()
+      receive_ts = Time.nanos()
```

Comment:
Why this change?

Reply:
I don't recall, but I'll restore the `Wallclock` version.